cpp/src/arrow/compute/exec/key_hash_test.cc

```diff
@@ -35,9 +35,11 @@ namespace compute {
 
 class TestVectorHash {
  private:
-  static Result<std::shared_ptr<BinaryArray>> GenerateUniqueRandomBinary(
+  template <typename Type, typename ArrayType = typename TypeTraits<Type>::ArrayType>
+  static Result<std::shared_ptr<ArrayType>> GenerateUniqueRandomBinary(
```

Comment:
Can we add a quick comment (or a template constraint) that `Type` must be a binary type (e.g. Binary / LargeBinary)?  Maybe it should be obvious but I didn't quite catch it the first glance through.

Reply:
Done.